README.md
# pravox.net
Website for Pravox Networks powered by Jekyll


news.md
---
layout: small
title: Redirecting to the Pravox Networks Blog
redirect_to:
  - https://medium.com/pravox-networks
---

<div id="main">
<div id="editable">
<div class="mz_component mz_wysiwyg mz_editable"> <div class="moze-wysiwyg-editor" markdown="1">
![Pravox Logo]({{site.logos|append:'pravox-networks-logo.png'}})
# Redirecting to the Pravox Networks Blog

</div></div></div>
<br class="clear">
</div>


js/mozlivead99.js
/* MozLive */

function mozLive(settings) {

    var base = this;

    /**
     * Default settings.
     */
    base.defaults = {
        src: null,          // Source component.
        dest: 'self',       // Destination component (for output).
        action: '',         // Component-specific action to execute.
        task: 'replace',    // Post-task for the outpot (append, replace).
        tasktarget: null,   // Post-task specific target (jQuery selector).
        parameters: {},     // Additional parameters for the action.
        errors: {
            maintenance: 'We can not process your request right now. Please try again later.'
        },
    };

    base.processResult = function (response) {

        // Execute callback

        if (typeof base.options.parameters.callback !== 'undefined') {
            base.options.parameters.callback(response);
        }

        // Is it a refresh task?

        if (base.options.task === 'refresh') {
            window.location.href = '/m/refresh/';
            return;
        }

        // Is it a redirect task?

        if (base.options.task === 'redirect') {
            if (typeof base.options.parameters.href !== 'undefined') {
                window.location.href = base.options.parameters.href;
            }
            return;
        }

        if (base.options.task == 'redirect-response') {
            window.location.href = response;
            return;
        }

        if (base.options.task == 'replace-html') {
            var newDoc = document.open('text/html');
            newDoc.write(response);
            newDoc.close();
            return;
        }

        // It should be replace or append task.

        var updatable = null;
        var updatableJq = null;

        if (base.options.dest !== null && base.options.dest !== 'self') {
            updatable = base.options.dest;
        }
        if (base.options.dest === 'self') {
            updatable = base.options.src;
        }

        if (updatable != null) {
            if (typeof updatable.id !== 'undefined') {
                updatableJq = $('[data-cid="' + updatable.id + '"]');
            }
            else if (typeof updatable.name !== 'undefined') {
                updatableJq = $('[data-name="' + updatable.name + '"]');
            }
        }

        if (updatableJq != null) {
            switch (base.options.task)
            {
                case 'replace':
                    if (base.options.tasktarget == null) {
                        $(updatableJq).replaceWith(response);
                    }
                    else {
                        $(updatableJq).find(base.options.tasktarget).replaceWith(response);
                    }
                    break;
                    
                case 'append':
                    $(updatableJq).append(response);
                    break;
            }
        }

    };

    /**
     * Executes the function.
     */
    base.run = function () {

        var base = this;

        $.ajax({
            url: '/m/mozlive/',
            type: 'post',
            data: {
                action: base.options.action,
                url: window.location.pathname,
                src: base.options.src,
                parameters: base.options.parameters,
            },
            success: function (result) {
                if (result.error == true) {
                    if (result.reason == 'maintenance') {
                        alert(base.options.errors.maintenance);
                    }
                }
                else {
                    base.processResult(result);
                }
            }
        });

    };

    // Runs the MozLive.

    base.options = $.extend({}, this.defaults, settings);
    base.run();

    return false;
}

/* End */

js/designfxad99.js
$(document).ready(function () {

    // Opens Gallery Pictures in a Fancybox.

    if (!$('body').hasClass('backend')) {
        var margin = 70;
        var show_arrows = true;
        var inside_close = false;
        if (isSmallTouchDevice()) {
            margin = 5;
            show_arrows = false;
            inside_close = true;
        }
        $('ul.moze-gallery.pictures li a, .mz_catalog a.fancy').fancybox({
            padding: 0,
            margin: margin,
            arrows: show_arrows,
            openEffect: 'none',
            closeEffect: 'none',
            // make prettier
            afterShow: function() {
                $('a.fancybox-prev').css('left', '-60px');
                $('a.fancybox-next').css('right', '-60px');
                if (inside_close) {
                    $('a.fancybox-close').css('right', '-3px');
                    $('a.fancybox-close').css('top', '-3px');
                }

            }
        });

        // Removes overlay if it is empty
        var olay = $('#bigbar-overlay')
        if (olay.length) {
            if ($.trim(olay.find('.moze-wysiwyg-editor:visible').text()) == '') {
                olay.css('background-color', 'transparent');
            }
        }
    }

    // Adjusts the title font size for images.

    var adjustFont = function () {
        var fontSize = $('.moze-gallery-overlay').width();
        var maxSize = parseInt($('body').css('font-size'));
        fontSize = Math.min(fontSize * 0.09, maxSize);
        $('.moze-gallery-overlay').css('font-size', fontSize);
    };

    adjustFont();
    $(window).resize(function () {
        adjustFont();
    });

    // Add Google Maps.

    function processMaps() {
        if ($('.moze-maps').length > 0) {
            var doc_lang = $('body').attr('lang');
            if (!doc_lang) {
                doc_lang = 'en';
            }
            $.getScript("http://maps.google.com/maps/api/js?v=3.exp&sensor=false&callback=MapApiLoaded&language=" + doc_lang, function () { });
        }
    }

    if (!$('body').hasClass('backend')) {
        processMaps();
    }

    // Loads MozBanner.

    if (!$('body').hasClass('backend')) {
        $('.mz_banner').mozbannerplay({});
    }

});

// Mobile detection

/* Detect mobiles */
function isSmallTouchDevice() {

    function isTouchDevice() {
      return 'ontouchstart' in window // works on most browsers
          || 'onmsgesturechange' in window; // works on ie10, mostly
    };

    return (isTouchDevice() && (window.screen.width < 480 || window.screen.height < 480));
}

// Google Maps stuff.

function initializeMap(jqelem) {
    geocoder = new google.maps.Geocoder();
    var latlng = new google.maps.LatLng(jqelem.data('lat'), jqelem.data('lng'));
    var mapOptions = {
        center: latlng,
        zoom: jqelem.data('zoom'),
        mapTypeId: google.maps.MapTypeId.ROADMAP,
        streetViewControl: true
    }
    map = new google.maps.Map(jqelem.get(0), mapOptions);
    map.setCenter(latlng);
    var marker = new google.maps.Marker({
        map: map,
        position: latlng
    });
    return map;
}

function MapApiLoaded() {
    $('.moze-maps').each(function () {
        initializeMap($(this));
    });
}

// Internal Mozello stuff

if (window.addEventListener) {
    window.addEventListener("message", receiveMessage, false);
}


function isExternalLinkOrImage(url) {
    var general_urls = new RegExp("(http|https)\:\/\/[^\.]+.[^\.]+", "g");
    var exclude_urls = new RegExp("^(http|https)\:\/\/(www\.)?(mozello|youtube|facebook|twitter)\.[a-z]{2,3}(\/|$)", "g");
    var content_urls = new RegExp("^(http|https)\:\/\/site\-[0-9]+\.mozfiles\.com", "g");
    if (url.match(general_urls) && !url.match(exclude_urls) && !url.match(content_urls)) {
        return true;
    }
    return false;
}

function receiveMessage(event)
{
    if (event.origin == "http://www.mozello.com") {
        if (event.data == 'highlight-links') {
            $('.mz_editable img').each(function () {
                var url = $(this).attr('src').trim();
                if (isExternalLinkOrImage(url)) {
                    $(this).css('border', '4px dotted blue');
                }
            });
            $('.mz_editable a').each(function () {
                var url = $(this).attr('href').trim();
                if (isExternalLinkOrImage(url)) {
                    $(this).css('border-bottom', '4px dotted red');
                    $(this).find('img').css('border-bottom', '4px dotted red');
                }
            });
        }
    }
}

/* End */

js/sharebuttons/sharebuttonseccb.js
/*

    Simple social buttons for Mozello
    Written by Karlis Blumentals
    used some parts by Artem Sapegin

*/

function simpleShare(settings) {

    var base = this;

    base.defaults = {
        url: window.location.href.replace(window.location.hash, ''),
        title: document.title,
        media: $('meta[property="og:image"]').length ? $('meta[property="og:image"]').attr('content') : '',
        showLabels: true,
        showCount: true
    };

    base.options = $.extend({}, base.defaults, settings);

    var services = {
        facebook: {
            counterFunction: function(elem) {
                var apiUrl = 'https://api.facebook.com/method/links.getStats?urls={url}&format=json';
                var url = makeUrl(apiUrl, {
                    url: base.options.url
                });
                $.getJSON(url, function(data) {
                    addShareCount(elem, data[0].total_count);
                });
            },
            popupUrl: 'https://www.facebook.com/sharer/sharer.php?u={url}',
            popupWidth: 600,
            popupHeight: 500
        },
        twitter: {
            popupUrl: 'https://twitter.com/intent/tweet?url={url}&text={title}',
            popupWidth: 600,
            popupHeight: 450
        },
        draugiem: {
            counterFunction: function(elem) {
                var apiUrl = 'https://www.draugiem.lv/say/ext/like_count.php?callback=?&url={url}';
                var url = makeUrl(apiUrl, {
                    url: base.options.url
                });
                $.getJSON(url, function(data) {
                    addShareCount(elem, data.count);
                });
            },
            popupUrl: 'https://www.draugiem.lv/say/ext/add.php?title={title}&link={url}&_n=1',
            popupWidth: 530,
            popupHeight: 400
        },
        vkontakte: {
            counterFunction: function(elem) {
                var apiUrl = 'https://vk.com/share.php?act=count&url={url}&index=1&callback=?';
                var url = makeUrl(apiUrl, {
                    url: base.options.url
                });
                // call api
                $.getJSON(url, function(data) {});
                // create VK object
                if (!window.VK) window.VK = {};
                window.VK.Share = {
                    count: function(idx, number) {
                        addShareCount(elem, number);
                    }
                }
            },
            popupUrl: 'https://vk.com/share.php?url={url}&title={title}',
            popupWidth: 550,
            popupHeight: 330
        },
        gplus: {
            popupUrl: 'https://plus.google.com/share?url={url}',
            popupWidth: 550,
            popupHeight: 500
        },
        pinterest: {
            counterFunction: function(elem) {
                var apiUrl = 'https://api.pinterest.com/v1/urls/count.json?callback=pinterestCounter&url={url}';
                var url = makeUrl(apiUrl, {
                    url: base.options.url
                });
                // call api
                $.getScript(url, function(data) {});
                // create pinterest callback
                window.pinterestCounter = function(json) {
                    addShareCount(elem, json.count);
                }
            },
            popupUrl: 'https://pinterest.com/pin/create/button/?url={url}&media={media}&description={title}',
            popupWidth: 630,
            popupHeight: 270
        }
    };

    function initButtons() {
        $('.ssb-container > a').each(function(){
            var service = $(this).data('service');
            var network = services[service];
            var name = $(this).attr('title');
            $(this).addClass('flat-' + service);
            if (base.options.showLabels) {
                $('<span class="ssb-label">' + name + '</span>').appendTo(this);
            }
            if (base.options.showCount) {
                $('<span class="ssb-counter"></span>').appendTo(this);
                // Share count
                if (network.counterFunction) {
                    network.counterFunction(this);
                }
            }
            // Sharing
            $(this).click(function(event) {
                event.preventDefault();
                var url = makeUrl(network.popupUrl, {
                    url: base.options.url,
                    title: base.options.title,
                    media: base.options.media
                });
                openPopup(url, this, network, {
                    width: network.popupWidth,
                    height: network.popupHeight
                });
                return false;
            });
        });
    }

    function addShareCount(elem, count) {
        if (count > 0) {
            $(elem).find('.ssb-counter').html(count);
        }
    }

    function makeUrl(url, context) {
        return template(url, context, encodeURIComponent);
    }

    function template(tmpl, context, filter) {
        return tmpl.replace(/\{([^\}]+)\}/g, function(m, key) {
            // If key doesn't exists in the context we should keep template tag as is
            return key in context ? (filter ? filter(context[key]) : context[key]) : m;
        });
    }

    function openPopup(url, elem, network, params) {
        var left = Math.round(screen.width/2 - params.width/2);
        var top = 0;
        if (screen.height > params.height) {
            top = Math.round(screen.height/3 - params.height/2);
        }
        var win = window.open(url, 'popup', 'left=' + left + ',top=' + top + ',' +
           'width=' + params.width + ',height=' + params.height + ',personalbar=0,toolbar=0,scrollbars=1,resizable=1');
        if (win) {
            win.focus();
            var timer = setInterval($.proxy(function() {
                    if (!win.closed) return;
                    clearInterval(timer);
                    if (base.options.showCount) {
                        // Update count
                        if (network.counterFunction) {
                            network.counterFunction(elem);
                        }
                    }
            }, this), 1000);
        } else {
            location.href = url;
        }
    }

    function loadRelativeCss(jssrc, cssfile) {
        var src = jssrc.substring(0, jssrc.lastIndexOf("\/") + 1);
        var cssLink = $('<link rel="stylesheet" type="text/css" href="' + src + cssfile + '">');
        $("head").append(cssLink);
    };

    loadRelativeCss(sharebutton_src, 'sharebuttons.css');
    initButtons();

    return false;

}

// For dynamic CSS loading - this must be global
var scripts = document.getElementsByTagName("script"),
sharebutton_src = scripts[scripts.length-1].src;

js/main.js
function mozLive(e){var t=this
return t.defaults={src:null,dest:"self",action:"",task:"replace",tasktarget:null,parameters:{},errors:{maintenance:"We can not process your request right now. Please try again later."}},t.processResult=function(e){if(void 0!==t.options.parameters.callback&&t.options.parameters.callback(e),"refresh"===t.options.task)return void(window.location.href="/m/refresh/")
if("redirect"===t.options.task)return void(void 0!==t.options.parameters.href&&(window.location.href=t.options.parameters.href))
if("redirect-response"==t.options.task)return void(window.location.href=e)
if("replace-html"==t.options.task){var n=document.open("text/html")
return n.write(e),void n.close()}var a=null,o=null
if(null!==t.options.dest&&"self"!==t.options.dest&&(a=t.options.dest),"self"===t.options.dest&&(a=t.options.src),null!=a&&(void 0!==a.id?o=$('[data-cid="'+a.id+'"]'):void 0!==a.name&&(o=$('[data-name="'+a.name+'"]'))),null!=o)switch(t.options.task){case"replace":null==t.options.tasktarget?$(o).replaceWith(e):$(o).find(t.options.tasktarget).replaceWith(e)
break
case"append":$(o).append(e)}},t.run=function(){var e=this
$.ajax({url:"/m/mozlive/",type:"post",data:{action:e.options.action,url:window.location.pathname,src:e.options.src,parameters:e.options.parameters},success:function(t){1==t.error?"maintenance"==t.reason&&alert(e.options.errors.maintenance):e.processResult(t)}})},t.options=$.extend({},this.defaults,e),t.run(),!1}function isSmallTouchDevice(){function e(){return"ontouchstart"in window||"onmsgesturechange"in window}return e()&&(window.screen.width<480||window.screen.height<480)}function initializeMap(e){geocoder=new google.maps.Geocoder
var t=new google.maps.LatLng(e.data("lat"),e.data("lng")),n={center:t,zoom:e.data("zoom"),mapTypeId:google.maps.MapTypeId.ROADMAP,streetViewControl:!0}
map=new google.maps.Map(e.get(0),n),map.setCenter(t)
new google.maps.Marker({map:map,position:t})
return map}function MapApiLoaded(){$(".moze-maps").each(function(){initializeMap($(this))})}function isExternalLinkOrImage(e){var t=RegExp("(http|https)://[^.]+.[^.]+","g"),n=RegExp("^(http|https)://(www.)?(mozello|youtube|facebook|twitter).[a-z]{2,3}(/|$)","g"),a=RegExp("^(http|https)://site-[0-9]+.mozfiles.com","g")
return!e.match(t)||e.match(n)||e.match(a)?!1:!0}function receiveMessage(e){"http://www.mozello.com"==e.origin&&"highlight-links"==e.data&&($(".mz_editable img").each(function(){var e=$(this).attr("src").trim()
isExternalLinkOrImage(e)&&$(this).css("border","4px dotted blue")}),$(".mz_editable a").each(function(){var e=$(this).attr("href").trim()
isExternalLinkOrImage(e)&&($(this).css("border-bottom","4px dotted red"),$(this).find("img").css("border-bottom","4px dotted red"))}))}$(document).ready(function(){function e(e){return e.replace(/^\//,"").replace(/(index|default).[a-zA-Z]{3,4}$/,"").replace(/\/$/,"")}function t(e){for(var t=0,n=arguments.length;n>t;t++){var a=arguments[t],o=$(a)
if(o.scrollTop()>0)return a
o.scrollTop(1)
var r=o.scrollTop()>0
if(o.scrollTop(0),r)return a}return[]}var n=e(location.pathname),a=t("html","body")
$("a[href*=#]").each(function(){var t=e(this.pathname)||n
if(n==t&&(location.hostname==this.hostname||!this.hostname)&&this.hash.replace(/#/,"")){var o=$(this.hash),r=this.hash
if(r){var i=o.offset().top
$(this).click(function(e){e.preventDefault(),$(a).animate({scrollTop:i},400,function(){location.hash=r})})}}})}),$(document).ready(function(){function e(){if($(".moze-maps").length>0){var e=$("body").attr("lang")
e||(e="en"),$.getScript("http://maps.google.com/maps/api/js?v=3.exp&sensor=false&callback=MapApiLoaded&language="+e,function(){})}}if(!$("body").hasClass("backend")){var t=70,n=!0,a=!1
isSmallTouchDevice()&&(t=5,n=!1,a=!0),$("ul.moze-gallery.pictures li a, .mz_catalog a.fancy").fancybox({padding:0,margin:t,arrows:n,openEffect:"none",closeEffect:"none",afterShow:function(){$("a.fancybox-prev").css("left","-60px"),$("a.fancybox-next").css("right","-60px"),a&&($("a.fancybox-close").css("right","-3px"),$("a.fancybox-close").css("top","-3px"))}})
var o=$("#bigbar-overlay")
o.length&&""==$.trim(o.find(".moze-wysiwyg-editor:visible").text())&&o.css("background-color","transparent")}var r=function(){var e=$(".moze-gallery-overlay").width(),t=parseInt($("body").css("font-size"))
e=Math.min(.09*e,t),$(".moze-gallery-overlay").css("font-size",e)}
r(),$(window).resize(function(){r()}),$("body").hasClass("backend")||e(),$("body").hasClass("backend")||$(".mz_banner").mozbannerplay({})}),window.addEventListener&&window.addEventListener("message",receiveMessage,!1)
